service/controller/app/v1/resource/tiller/resource.go

```diff
@@ -0,0 +1,96 @@
+package tiller
+
+import (
+	"context"
+
+	"github.com/giantswarm/errors/tenant"
+	"github.com/giantswarm/helmclient"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	"github.com/giantswarm/operatorkit/controller/context/reconciliationcanceledcontext"
+	"github.com/giantswarm/tenantcluster"
+)
+
+const (
+	Name = "tillerv1"
+)
+
+// Config represents the configuration used to create a new tiller resource.
+type Config struct {
+	// Dependencies.
+	Logger micrologger.Logger
+}
+
+type Resource struct {
+	// Dependencies.
+	logger micrologger.Logger
+}
+
+// New creates a new configured tiller resource.
+func New(config Config) (*Resource, error) {
+	if config.Logger == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.Logger must not be empty", config)
+	}
+
+	r := &Resource{
+		// Dependencies.
+		logger: config.Logger,
+	}
+
+	return r, nil
+}
+
+func (r Resource) Name() string {
+	return Name
+}
+
+func (r *Resource) ensureTillerInstalled(ctx context.Context, helmClient helmclient.Interface) error {
+	r.logger.LogCtx(ctx, "level", "debug", "message", "ensuring tiller is installed")
+
+	values := []string{
+		"spec.template.spec.priorityClassName=giantswarm-critical",
+		"spec.template.spec.tolerations[0].effect=NoSchedule",
+		"spec.template.spec.tolerations[0].key=node-role.kubernetes.io/master",
+		"spec.template.spec.tolerations[0].operator=Exists",
+	}
+
+	var err error
+
+	err = helmClient.EnsureTillerInstalledWithValues(ctx, values)
+	if tenantcluster.IsTimeout(err) {
+		r.logger.LogCtx(ctx, "level", "debug", "message", "timeout fetching certificates")
+
+		// A timeout error here means that the app-operator certificate
```

Comment:
These comments are from cluster-operator, are these okay?

Reply:
We can remove this case because we don't use tenantcluster. It was replaced by kubeconfig and the clients resource handles creating the helmclient.

Other cases are fine.